addons/web/static/src/webclient/actions/action_service.js

```diff
@@ -536,9 +536,10 @@ function makeActionManager(env) {
                     dialogCloseResolve();
                 }
             }
-            onHistoryBack() {
+            async onHistoryBack() {
                 const previousController = controllerStack[controllerStack.length - 2];
-                if (previousController && !dialogCloseProm) {
+                if (previousController && !dialog.remove) {
+                    await dialogCloseProm;
```

Comment:
not sure we should wait for it, as if we do, we go back to B for a brief moment. If we don't wait, we may cancel the rendering of B and directly go back to A, isn't it?

Reply:
fair enough